### ISSUE TITLE
Feature request: Ability to restart container

### ISSUE THREAD

**Issue by cwkang1998** [testcontainers/testcontainers-node#340]
As of right now, there is no possible way provided to restart started containers. I would like to see a exposed method to restart a given container.

The use case for this may be specific to me (or may not be), but generally there is a need for me to change the configuration file and push for a restart after the config change. As of right now, I am trying to work it through via the use of `Dockerode`, which this is built on top of, but found no success.

I am willing to implement this myself for this feature, but would like to know the project maintainer's thought on this feature, considering it does not exists in `testcontainer-java`.

Related: https://github.com/testcontainers/testcontainers-java/issues/606

**Comment by cruxicheiros** [comment]
@cristianrgreco is there a straightforward way to wait for a particular log message after restarting a process in a container? I have this problem too and that's what stops me from doing that - if I create a new container I get `Wait.forLogMessage` but I have to wait for my container to start, which takes a long time because it needs to build everything from scratch due to being set up for hot reload. Meanwhile, if I restart the process I can't wait for the container to actually be ready to recieve messages particularly easily.

**Comment by cristianrgreco** [comment]
Hi @cruxicheiros, yes as per the README you can stream the logs of a running container:

```javascript
const stream = await container.logs();
stream
    .on("data", line => console.log(line))
    .on("err", line => console.error(line))
    .on("end", () => console.log("Stream closed"));
```

I guess you want something like:

```javascript
await new Promise(resolve => {
  const stream = await container.logs();
  stream.on("data", line => {
    if (line === "PROCESS READY") {
      resolve();
    }
  })
});
```

**Comment by toefel18** [comment]
I would really like this feature as well, I want to test kafka auto-healing behavior, but as of right now when I do a stop() and start() I get new mappings. I want exactly the same host:port to test this behaviour. Mapping the ports to something I explicitly want is also not allowed/recommended anymore. 

What is the correct way of testing such a feature?

**Comment by cristianrgreco** [comment]
Released in [8.12.0](https://github.com/testcontainers/testcontainers-node/releases/tag/v8.12.0), thanks @cwkang1998
